internal/controller/server_controller.go

```diff
@@ -328,11 +328,13 @@ func (r *ServerReconciler) handleReservedState(ctx context.Context, log logr.Log
 	}
 	log.V(1).Info("Server boot configuration is ready")
 
-	if err := r.pxeBootServer(ctx, log, server); err != nil {
-		return false, fmt.Errorf("failed to boot server: %w", err)
+	//TODO: handle working Reserved Server that was suddenly powered off but needs to boot from disk
+	if server.Status.PowerState == metalv1alpha1.ServerOffPowerState {
```

Comment:
I am not sure I understand this flow here. 🤔

Reply:
I would like to requeue `Reserved` server to poll for the power state but I don't want to set PXE boot again if it's already running and powered on. Given that we now poll for power more often, we ensure that for powered on `Server` boot order is not set. 

If server switches from `Available` (power state off) to `Reserved` then we set PXE boot. Maybe there is a better way to achieve this though.

Reply:
Fair point.